src/odemis/gui/cont/tabs.py

```diff
@@ -2201,91 +2207,152 @@ def __init__(self, name, button, panel, main_frame, main_data):
+            # Event binding for tab controls
+            panel.btn_switch_loading.Bind(wx.EVT_BUTTON, self._on_switch_btn)
+            panel.btn_switch_imaging.Bind(wx.EVT_BUTTON, self._on_switch_btn)
+            panel.btn_switch_milling.Bind(wx.EVT_BUTTON, self._on_switch_btn)
+            panel.btn_switch_coating.Bind(wx.EVT_BUTTON, self._on_switch_btn)
+            panel.stage_align_btn_p_aligner_x.Bind(wx.EVT_BUTTON, self._on_aligner_btn)
+            panel.stage_align_btn_m_aligner_x.Bind(wx.EVT_BUTTON, self._on_aligner_btn)
+            panel.stage_align_btn_p_aligner_y.Bind(wx.EVT_BUTTON, self._on_aligner_btn)
+            panel.stage_align_btn_m_aligner_y.Bind(wx.EVT_BUTTON, self._on_aligner_btn)
+            panel.stage_align_btn_p_aligner_z.Bind(wx.EVT_BUTTON, self._on_aligner_btn)
+            panel.stage_align_btn_m_aligner_z.Bind(wx.EVT_BUTTON, self._on_aligner_btn)
+            panel.btn_cancel.Bind(wx.EVT_BUTTON, self._on_cancel)
+
+            # Determine and show current position of the stage
+            self._current_position, self._target_position = None, None
+            self._enable_movement_controls()
+            if self._current_position in self.position_btns.keys():
+                pos_button = next(button for pos, button in self.position_btns.items() if pos == self._current_position)
+                self._toggle_switch_buttons(pos_button)
+            self._show_cancel_warning_msg(None)
+
+        elif self._role == 'meteor':
+            self._stage  = self.tab_data_model.main.stage_bare
+            # start and end position are used for the gauge progress bar
+            self._start_pos = self._end_pos = None
+            # Show current position of the stage via the progress bar
+            self._stage.position.subscribe(self._update_progress_bar, init=False)
+            self._stage.position.subscribe(self._on_stage_pos)
+            # metadata 
+            # check for the metadata of meteor stage positions
+            if not model.MD_SAMPLE_CENTERS in self._stage.getMetadata():
+                raise ValueError("The stage misses the 'SAMPLE_CENTERS' metadata.")
+            if not model.MD_FAV_POS_DEACTIVE in self._stage.getMetadata():
+                raise ValueError("The stage misses the 'FAV_POS_DEACTIVE' metadata.")
+            if not model.MD_SEM_IMAGING_RANGE in self._stage.getMetadata():
+                raise ValueError("The stage misses the 'SEM_IMAGING_RANGE' metadata.")
+            if not model.MD_FM_IMAGING_RANGE in self._stage.getMetadata():
+                raise ValueError("The stage misses the 'FM_IMAGING_RANGE' metadata.")
+            # Fail early when required axes are not found on the positions metadata
+            focuser = self.tab_data_model.main.focus
+            focus_md = focuser.getMetadata()
+            required_axes = {'z'}
+            for focuser_position in focus_md.values():
+                if not required_axes.issubset(focuser_position.keys()):
```

Comment:
Not all focuser metadata has to have z axis. Only POS_ACTIVE & POS_DEACTIVE positions. So make sure to only check for these metadata.